doc: add 'Release Phases' documentation (#517)
nodejs/Release: README.md

The Release schedule is available also as a [JSON][] file.

### Release Phases

There are three phases that a Node.js release can be in: 'Current', 'Active

> Long Term Support', and 'Maintenance'. Odd-numbered release lines are not

Reviewer: ```suggestion
Long Term Support (LTS)', and 'Maintenance'. Odd-numbered release lines are not
```

The Release schedule is available also as a [JSON][] file.

### Release Phases

There are three phases that a Node.js release can be in: 'Current', 'Active
Long Term Support (LTS)', and 'Maintenance'. Odd-numbered release lines are not

> promoted to LTS - they will not go through the 'Active LTS' or 'Maintenance'

Reviewer 2: I thought that odd number releases went through a short maintenance period? is that no longer true?

Author: It is true according to the `schedule.json`- I think we've been purposely vague about it in the past. We can change that - maybe something like "release lines may receive critical fixes shortly after going EOL at the discretion of the Release team"? But I am not sure which is the best approach personally.

phases.

 * Current - Should incorporate most of the non-major (non-breaking)
 changes that land on `nodejs/node` master branch.
 * Active LTS - New features, bug fixes, and updates that have been audited by
 the LTS team and have been determined to be appropriate and stable for the
 release line.
 * Maintenance - Critical bug fixes and security updates. New features may be
 added at the discretion of the LTS team - typically only in cases where
 the new feature supports migration to later release lines.

Changes required for critical security and bug fixes may lead to *semver-major*

> changes landing within an LTS stream, such situations will be rare and will

Reviewer 2: this could happen in LTS or Current afaik
```suggestion
changes landing within a release stream, such situations will be rare and will
```

The Release schedule is available also as a [JSON][] file.

### Release Phases

> There are three phases that a Node.js release can be in: 'Current', 'Active Long-term Support', and 'Maintenance'. Odd-numbered release lines are not promoted to LTS - they will not go through the 'Active LTS' or 'Maintenance' phases. 

Reviewer 3: Nit: it would be good to keep the 80 character limit as in the rest of the file.

## Release Plan

New *semver-major* releases of Node.js are branched from `master` every six
months. New even-numbered versions are released in April and odd-numbered
versions in October.

In coordination with a new *odd-numbered* major release, the previous
*even-numbered* major version will transition to Long Term Support. The
transition to Long Term Support will happen in a *semver-minor* release and can

>  happen either before or after the new major version is released.

Reviewer 2: We may want to open a follow up issue to track this, if I recall this is the exact thing that caused the "two current releases" thing.

As we will also be transitioning Active LTS releases to maintenance at the same time now we may want to have slightly more robust / clear policy here.

To be clear, not blocking... this should be a follow up.

Author: Agreed - I think https://github.com/nodejs/Release/issues/495 had some of the conversation tied to it
